### PR TITLE
Add HTTP Bearer token to Davix wrapper from config or env

### PR DESCRIPTION
**Pull request by dciangot**
Issue #6802

### CODE REVIEW

**Comment by jblomer** [comment]
@phsft-bot build

File: net/davix/src/TDavixFile.cxx [hunk 1]
```diff
@@ -417,6 +417,17 @@ void TDavixFileInternal::parseConfig()
    if (gDebug > 0)
       Info("parseConfig", "Setting CAcheck to %s", ((ca_check_local) ? ("true") : ("false")));
 
+   std::string prefix = "Bearer ";
+   // setup Bearer token auth
+   env_var = gEnv->GetValue("Davix.HTTP.Token", getenv("HTTP_TOKEN"));
+      if (gDebug > 0)
+         Info("parseConfig", "Using Bearer TOKEN: %s", env_var);
```
Comment (jblomer) [inline net/davix/src/TDavixFile.cxx:424]: I think this breaks if `env_var` is `nullptr`

File: net/davix/src/TDavixFile.cxx [hunk 2]
```diff
@@ -417,6 +417,17 @@ void TDavixFileInternal::parseConfig()
    if (gDebug > 0)
       Info("parseConfig", "Setting CAcheck to %s", ((ca_check_local) ? ("true") : ("false")));
 
+   std::string prefix = "Bearer ";
+   // setup Bearer token auth
+   env_var = gEnv->GetValue("Davix.HTTP.Token", getenv("HTTP_TOKEN"));
+      if (gDebug > 0)
```
Comment (jblomer) [inline net/davix/src/TDavixFile.cxx:423]: There is a whitespace issue here

**Review by jblomer — CHANGES_REQUESTED** [review]
In principal looks good to me.  I'd appreciate @oshadura and @pcanal taking a look, too.

I think we should try and add a unit test.  We can setup a directory on https://root.cern/files with a known bearer authorization token for testing.  Can you advise on the Apache configuration?

File: net/davix/src/TDavixFile.cxx (hunk 2 above)
Comment (jblomer) [inline net/davix/src/TDavixFile.cxx:423]: This `if` statement can be removed I think.  Logging is covered inside the `if (env_var)` block.

File: net/davix/src/TDavixFile.cxx (hunk 1 above)
Comment (bbockelm) [reply net/davix/src/TDavixFile.cxx:424]: I would suggest using `BEARER_TOKEN` instead of `HTTP_TOKEN`; the former is what `gfal2` uses.

Comment (bbockelm) [reply net/davix/src/TDavixFile.cxx:424]: `TOKEN` is not an acronym.  For the debug messages, just use `token`.

This is also a security credential - I would strongly suggest to redact it from debug output lest these get copy/pasted into the root forum.

**Comment by bbockelm** [comment]
@dciangot - the WLCG has published a specification for how applications should discover bearer tokens from their environment:

https://github.com/WLCG-AuthZ-WG/bearer-token-discovery/blob/master/specification.md

This would be the first step along those lines ... unless there's a strong reason not to, it might be a good idea to implement the full spec.

**Comment by bbockelm** [comment]
@dciangot - have you had a chance to look at this and/or think about my suggestion of using the standard token discovery document?

**Comment by dciangot** [comment]
@bbockelm the use of the standard token discovery sounds good to me, but haven't got time to look at that in practice yet. 
Unfortunately this is not a top priority from my side atm, so it's sort of best effort.
Nevertheless if you find it anyway useful or urgent I can try to enhance it a bit.

**Comment by oshadura** [comment]
@dciangot I think we can close this PR in favor of merged #7068?

**Comment by dciangot** [comment]
yes, that's ok for me

**Pull request closed without merging**